ls1intum/Artemis
# `General`: Increase interactivity of tutor effort distribution chart

### Checklist
#### General


#### Client

### Motivation and Context


This PR changes the appearance and interactivity of the tutor effort distribution chart

### Description

- **Positioning**: The chart as well as the explanation text if no information can be displayed are now centered on the page. Furthermore, the recalculation button is moved to the right side.
- **Help icon**: A help icon next to the chart title provides a tooltip that is explaining what data is displayed there
- **Median**: The median of the distribution is now displayed in the table above the chart. If the bar representing the interval the median is contained in is not 0, this bar is highlighted in a darker blue and a corresponding color legend appears next to the table header of the median.
- **Enhanced tooltips**: The tooltips are formulated more clearly and now provide an explanation of what is displayed by the selected bar. Additionally, the tooltip contains the median of assessed submissions from tutors that fell in the corresponding interval.
- **Clickable bars**: By clicking on a chart bar, the user is redirected to the exercise assessment dashboard

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Text exercise with existing tutor efforts

**Important**: There seems to be problems with the computation of the effort statistics. These are not due to my changes, as they also appear on develop (at least for me). If no data is shown to you even though you assessed submissions, you might have to do it again (or do it this way directly):

- After submitting an assessment, do not press "assess next submission". Instead return to the assessment dashboard and start the next assessment from there.

If this does not work for you as well, inspect [this distribution](https://artemistest.ase.in.tum.de/course-management/481/text-exercises/13863/tutor-effort-statistics) on TS1 (e.g. with Test user 12). This is obviously not a long-term solution but might allow you to test this PR. 

1. Log in to Artemis
2. Navigate to Course Administration
3. Click on the exercise name and select "Tutor effort statistics"
4. Inspect the chart:
     - Check that the highlighting of the median works as described above
     - Check that the tooltips are understandable and well structured
     - Check that the explanation of the help icon is understandable
     - Check that you are delegated to the assessment dashboard if you click on a bar chart

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






| File name | Branch coverage (%) | Line coverage (%) |
| -- | -- | -- | 
| `tutor-effort-statistics.component.ts` | 76.47 | 95.23 |

### Screenshots


![image](https://user-images.githubusercontent.com/80622272/153641729-3363f69b-92da-4a32-ad2b-a6ad58bb9676.png)
